hooks/kubernetes_installer.py

```diff
@@ -20,43 +18,44 @@ def __init__(self, arch, version, kubernetes_file):
                         'kubectl': 'kubectl'}
         self.arch = arch
         self.version = version
-        self.kubernetes_file = kubernetes_file
+        self.kubernetes_file = path(kubernetes_file)
 
-    def install(self, output_dir='/opt/kubernetes'):
+    def install(self, output_dir=path('/opt/kubernetes')):
         """ Install kubernetes binary files from the tar file or gsutil. """
-        if os.path.isdir(output_dir):
-            # Remote old content to remain idempotent.
-            shutil.rmtree(output_dir)
-        kubernetes_repo = 'http://github.com/GoogleCloudPlatform/kubernetes'
-        git_clone = 'git clone {0} {1}'.format(kubernetes_repo, output_dir)
-        print(git_clone)
-        subprocess.check_call(git_clone.split())
+        bindir = output_dir / 'bin'
+        if bindir.exists():
+            for fp in bindir.files():
+                fp.remove()
+
+        bindir.makedirs_p()
 
-        output_dir = os.path.join(output_dir, 'bin')
-        # Create an output directory for binaries.
-        os.makedirs(output_dir)
+        kubernetes_repo = 'http://github.com/GoogleCloudPlatform/kubernetes'
+        repo_dir = output_dir / 'kubernetes/src'
```

Comment:
git cloning kubernetes creates a complex directory structure on to its own, I don't think we need the kubernetes/src here and will break the documentation that is written.

The way I read this creates /opt/kubernetes/kubernetes/src and that isn't what we want.